src/lib/Helpers/ParsingHelpers.cs

```diff
@@ -136,6 +138,46 @@ internal static SortedSet<string> GetOrderedHashSetOfString(JsonElement v)
             yield return keyValue;
         }
     }
+
+    internal static async Task<ReadResult> ParseOpenApiAsync(Uri openApiFileUri, bool inlineExternal, CancellationToken cancellationToken)
+    {
+        Stream stream = await GetStreamAsync(openApiFileUri, cancellationToken: cancellationToken);
+        return await ParseOpenApiAsync(stream, openApiFileUri, inlineExternal, cancellationToken);
+    }
+
+    internal static async Task<ReadResult> ParseOpenApiAsync(Stream stream, Uri openApiFileUri, bool inlineExternal, CancellationToken cancellationToken)
+    {
+        ReadResult result = await new OpenApiStreamReader(new OpenApiReaderSettings
+        {
+            LoadExternalRefs = inlineExternal,
+            BaseUrl = openApiFileUri
+        }
+        ).ReadAsync(stream, cancellationToken);
+
+        return result;
+    }
+
+    internal static async Task<Stream> GetStreamAsync(Uri uri, HttpMessageHandler? finalHandler = null, CancellationToken cancellationToken = default)
+    {
+        if (!uri.Scheme.StartsWith("http"))
+            throw new ArgumentException($"The input {uri} is not a valid url", nameof(uri));
+        try
+        {
+            finalHandler ??= new HttpClientHandler()
+            {
+                SslProtocols = System.Security.Authentication.SslProtocols.Tls12,
+            };
+            using var httpClient = new HttpClient(finalHandler)
```

Comment:
this might lead to port exhausting if the method is called frequently. Maybe we should rely on a singleton for the http client? (lazy static)

Reply:
Good call!